pulp_container/app/registry_api.py

```diff
@@ -442,41 +448,47 @@ def put(self, request, path, pk=None):
         digest = request.query_params["digest"]
         upload = models.Upload.objects.get(pk=pk, repository=repository)
 
-        if upload.sha256 == digest[len("sha256:") :]:
-            try:
-                artifact = Artifact(
-                    file=upload.file.name,
-                    md5=upload.md5,
-                    sha1=upload.sha1,
-                    sha256=upload.sha256,
-                    sha384=upload.sha384,
-                    sha512=upload.sha512,
-                    size=upload.file.size,
-                )
-                artifact.save()
-            except IntegrityError:
-                artifact = Artifact.objects.get(sha256=artifact.sha256)
-            try:
-                blob = models.Blob(digest=digest, media_type=models.MEDIA_TYPE.REGULAR_BLOB)
-                blob.save()
-            except IntegrityError:
-                blob = models.Blob.objects.get(digest=digest)
-            try:
-                blob_artifact = ContentArtifact(
-                    artifact=artifact, content=blob, relative_path=digest
-                )
-                blob_artifact.save()
-            except IntegrityError:
-                pass
+        chunks = models.BlobTemporaryUpload.objects.filter(upload=upload).order_by("offset")
+        chunks_files = map(lambda chunk: chunk.file, chunks)
+
+        with NamedTemporaryFile("ab") as temp_file:
```

Comment:
Isn't there a "turn this `PulpTemporaryFile` into an artifact primitive suitable for this?

Reply:
No, since I am constructing an Artifact object from a bunch of PulpTemporaryFile objects, like I did mention above.

Reply:
Sorry, i thought you assembled all the chunks into a new PTFile, but then again you can just create the Artifact directly.

Reply:
I cannot create an Artifact directly because it is not possible to open files in the append mode. I have to iterate through chunks, merge them, and then create an Artifact on a single write.

Reply:
Yeah, that is what i meant. I first thought you'd "upload" that (locally) assembled file into a new PTFile to make an artifact from it. But that step is clearly unnecessary, because you can create the artifact directly from that assembled file.